src/main/java/core/basesyntax/service/ProcessInputService.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.HandleTransactionStrategy;
+import core.basesyntax.service.strategy.HandleTransactionStrategyImpl;
+import core.basesyntax.service.strategy.TransactionHandler;
+import java.util.List;
+
+public class ProcessInputService {
+    public boolean parseInput(List<String> inputList) {
```

Comment:
Single Responsibility - currently, this method is responsible for parsing String -> FruitTransaction and choosing a strategy for subsequent processing - this is not correct